api/src/Repository/TopBook/TopBookDataRepository.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repository\TopBook;
+
+use App\Entity\TopBook;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+final class TopBookDataRepository implements TopBookDataInterface
+{
+    private const DATA_SOURCE = 'top-100-novel-sci-fi-fr.csv';
+    private const FIELDS_COUNT = 5;
+
+    private KernelInterface $kernel;
+
+    public function __construct(KernelInterface $kernel)
```

Comment:
Prefer injecting the projectDir param instead of the kernel:
```yaml
# config/services.yaml
services:
    _default:
        # ...
        bind:
            $projectDir: '%kernel.project_dir%'
```
```suggestion
    public function __construct(string $projectDir)
```